app/javascript/readingList/readingList.jsx

```diff
@@ -37,6 +37,12 @@ export class ReadingList extends Component {
     }, 1);
   }
 
+  componentWillUpdate(_, newState) {
+    const { readingListItems } = newState;
```

Comment:
I think the count should come from the `nbHits`  value inside the Algolia search's response: https://www.algolia.com/doc/api-reference/api-methods/search/#method-response-nbhits

Basically:

* `componentDidMount` calls Algolia's search, which returns `nbHits` 
* that value gets stored in the component state
* each time an item is archived the count is decreased by 1
* the other views already call `listSearch` which should override the items count as well, so that switching from one page to the other will have an accurate count

The limitation with the current approach is that it only shows the number of items in the current page (64 at most), not the accurate count